File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/converter/StringToFloatArrayConverter.java
package de.osthus.ambeth.converter;

import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.util.IDedicatedConverter;

public class StringToFloatArrayConverter implements IDedicatedConverter
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Override
	public Object convertValueToType(Class<?> expectedType, Class<?> sourceType, Object value, Object additionalInformation)
	{
		if (float[].class.equals(expectedType))
		{
			String[] split = StringToPatternConverter.splitPattern.split((String) value);
			float[] result = new float[split.length];
			for (int a = split.length; a-- > 0;)
			{
				result[a] = Float.parseFloat(split[a]);
			}
			return result;
		}
		StringBuilder sb = new StringBuilder();
		float[] array = (float[]) value;
		for (float item : array)
		{
			if (sb.length() > 0)
			{
				sb.append(StringToPatternConverter.splitPattern.pattern());
			}
			sb.append(item);
		}
		return sb.toString();
	}
}

File: ambeth/Ambeth.IoC/ambeth/converter/StringToFloatArrayConverter.cs
using De.Osthus.Ambeth.Log;
using De.Osthus.Ambeth.Util;
using System;
using System.Text;

namespace De.Osthus.Ambeth.Converter
{
    public class StringToFloatArrayConverter : IDedicatedConverter
    {
        [LogInstance]
        public ILogger Log { private get; set; }

        public Object ConvertValueToType(Type expectedType, Type sourceType, Object value, Object additionalInformation)
        {
            if (typeof(float[]).Equals(expectedType))
            {
                String[] split = StringToPatternConverter.splitPattern.Split((String)value);
                float[] result = new float[split.Length];
                for (int a = split.Length; a-- > 0; )
                {
                    result[a] = Single.Parse(split[a]);
                }
                return result;
            }
            StringBuilder sb = new StringBuilder();
            float[] array = (float[])value;
            foreach (float item in array)
            {
                if (sb.Length > 0)
                {
                    sb.Append(StringToPatternConverter.splitPattern.ToString());
                }
                sb.Append(item);
            }
            return sb.ToString();
        }
    }
}